staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -166,3 +166,56 @@ type TracingConfiguration struct {
 	// Defaults to 0.
 	SamplingRatePerMillion *int32
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// FlowControlConfig holds flow control parameters.
+type FlowControlConfig struct {
+	metav1.TypeMeta
+
+	// WorkEstimatorConfig holds work estimator parameters.
+	// +optional
+	*WorkEstimatorConfig
+}
+
+// WorkEstimatorConfig holds work estimator parameters.
+type WorkEstimatorConfig struct {
+	// ListWorkEstimatorConfig holds work estimator parameters related to list requests.
+	// +optional
+	ListWorkEstimatorConfig *ListWorkEstimatorConfig
+	// MutatingWorkEstimatorConfig holds work estimator
+	// parameters related to watches of mutating objects.
+	// +optional
+	MutatingWorkEstimatorConfig *MutatingWorkEstimatorConfig
+
+	// MaximumSeats is the maximum number of seats a request can occupy
+	//
+	// NOTE: work_estimate_seats_samples metric uses the value of maximumSeats
+	// as the upper bound, so when we change maximumSeats we should also
+	// update the buckets of the metric.
+	// +optional
+	MaximumSeats uint
```

Comment:
Is https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types relevant?  Arguably not, since we are concerned here only with the Go-language representation.

Reply:
This is caused by `WorkEstimate.{InitialSeats, FinalSeats}` ([src](https://github.com/kubernetes/kubernetes/blob/1c594e7e01a899807431c806cd11c1d27c885c9c/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/width.go#L33)) being `uint`. If we specify `MaximumSeats` here to be an `int` then we have to perform casting each time there is an assignment (e.g. [src](https://github.com/kubernetes/kubernetes/blob/1c594e7e01a899807431c806cd11c1d27c885c9c/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/list_work_estimator.go#L90)).
I'd leave it as `uint` for now but create an issue/pr for that to be changed everywhere to an `int`.

Reply:
Just changing to "int" would be only partial compliance with the guidelines I cited. I think the first question to answer is whether those guidelines --- which were written for API objects manipulated by clients written in a wide variety of programming languages --- are relevant to a config API (which, by its nature, is relevant only to the programming language of the program being configured).

Reply:
We can but this will cause the same issue as with changing the type `int` - see https://github.com/kubernetes/kubernetes/pull/111222#discussion_r928683481. If you believe it's a needed change I'll just apply it in this PR, please confirm.

Reply:
Moved to https://github.com/kubernetes/kubernetes/pull/111503.

Reply:
https://github.com/kubernetes/kubernetes/pull/111503 is merged. Rebased this PR to use `uint64`.

Reply:
Ugh.  I am sorry I raised the question.  I think the right answer is that the concerns driven by not knowing an API's client programming language are irrelevant here.